src/main/java/com/iota/iri/network/TransactionRequester.java

```diff
@@ -84,48 +82,49 @@ private boolean transactionsToRequestIsFull() {
 
 
     public Hash transactionToRequest(boolean milestone) throws Exception {
-        final long beginningTime = System.currentTimeMillis();
+        // determine which set of transactions to operate on
+        Set<Hash> primarySet = milestone ? milestoneTransactionsToRequest : transactionsToRequest;
+        Set<Hash> alternativeSet = milestone ? transactionsToRequest : milestoneTransactionsToRequest;
+        Set<Hash> requestSet = primarySet.size() == 0 ? alternativeSet : primarySet;
+
+        // determine the first hash in our set that needs to be processed
         Hash hash = null;
-        Set<Hash> requestSet;
-        if(milestone) {
-             requestSet = milestoneTransactionsToRequest;
-             if(requestSet.size() == 0) {
-                 requestSet = transactionsToRequest;
-             }
-        } else {
-            requestSet = transactionsToRequest;
-            if(requestSet.size() == 0) {
-                requestSet = milestoneTransactionsToRequest;
-            }
-        }
         synchronized (syncObj) {
+            // repeat while we have transactions that shall be requested
```

Comment:
For next time instead of adding comments you can just refactor the code to be more readable.
For example you could have wrote `!requestSize.isEmpty()` or if you want a null set to be considered as an empty set (in this specific example it won't ever be null) `CollectionUtils.isNotEmpty(requestSet)`